x-pack/platform/plugins/shared/fleet/server/services/epm/archive/extract.ts

```diff
@@ -53,7 +53,7 @@ export async function unzipBuffer(
 
     try {
       if (shouldReadBuffer && !shouldReadBuffer(path)) {
-        return onEntry({ path });
+        return await onEntry({ path });
```

Comment:
not sure if it could cause an issue but seems safer to await here, before reading the next entry similar to what's done when using the buffer